incubating/sonar-scanner-dotnet-linux/step.yaml

```diff
@@ -0,0 +1,172 @@
+          arguments:
+            SONAR_HOST_URL: https://sonarqube.codefresh.io
+            SONAR_PROJECT_KEY: myprojectkey
+            SONAR_LOGIN: myuser
+            SONAR_PASSWORD: mypassword
+            DOTNET_COMMANDS:
+              - docker build...
+            SONAR_ANALYSIS_PARAMETERS:
+              - sonar.projectDescription='myprojectdesc'
+              - sonar.log.level=TRACE
+spec:
+  arguments: |-
+    {
+      "definitions": {},
+      "$schema": "http://json-schema.org/draft-07/schema#",
+      "type": "object",
+      "additionalProperties": true,
+      "patterns": [],
+      "required": [
+        "SONAR_HOST_URL",
+        "SONAR_LOGIN",
+        "SONAR_PROJECT_KEY"
+      ],
+      "properties": {
+        "SONAR_HOST_URL": {
+          "type": "string",
+          "description": "Sonarqube Server Host URL."
+        },
+        "SONAR_HOST_URL": {
+          "type": "string",
+          "description": "Sonarqube Server Host URL."
+        },
+        "SONAR_LOGIN": {
+          "type": "string",
+          "description": "Sonarqube Username or Token."
+        },
+        "SONAR_PASSWORD": {
+          "type": "string",
+          "description": "Sonarqube Password if using Username for SONAR_LOGIN."
+        },
+        "SONAR_PROJECT_KEY": {
+          "type": "string",
+          "description": "Sonarqube Project Key."
+        },
+        "SONAR_PROJECT_NAME": {
+          "type": "string",
+          "description": "Sonarqube Project Name."
+        },
+        "SONAR_VERBOSE": {
+          "type": "boolean",
+          "description": "Sonarqube Project Key."
+        },
+        "SONAR_ANALYSIS_PARAMETERS": {
+          "type": "array",
+          "description": "Array of additional Sonarqube Analysis Parameters Ex. - sonar.projectDescription='mycustomdescription'. List: https://docs.sonarqube.org/latest/analysis/analysis-parameters/"
+        },
+        "REGISTRY": {
+          "type": "string",
+          "description": "Sonarqube dotnet scanner Docker Registry.",
+          "default": ""
+        },
+        "IMAGE": {
+          "type": "string",
+          "description": "Sonarqube dotnet scanner Docker image.",
+          "default": "codefreshplugins/sonar-scanner-dotnet-linux"
+        },
+        "IMAGE_TAG": {
+          "type": "string",
+          "description": "Sonarqube dotnet scanner Docker image tag.",
+          "default": "0.0.1"
+        },
+        "DOTNET_COMMANDS": {
+          "type": "array",
+          "description": "Commands to execute instead of the default scanning commands.",
+          "default": [
+            "dotnet build"
+          ]
+        },
+        "PRE_COMMANDS": {
+          "type": "array",
+          "description": "Commands to execute before scan command."
+        },
+        "POST_COMMANDS": {
+          "type": "array",
+          "description": "Commands to execute after scan command."
+        },
+        "WORKING_DIRECTORY": {
```

Comment:
Should this be a required field?

Reply:
Nope, if you look below this is just a way to specify a path deeper than the root of the checkout.  The checkout is default.